core/src/main/java/org/primefaces/extensions/component/clockpicker/ClockPickerRenderer.java

```diff
@@ -0,0 +1,200 @@
+        writer.endElement("span");
+        writer.endElement("span");
+        writer.endElement("div");
+    }
+
+    private void encodeScript(final FacesContext context, final ClockPicker clockPicker, final String value) throws IOException {
+        final WidgetBuilder wb = getWidgetBuilder(context);
+        final String clientId = clockPicker.getClientId(context);
+
+        wb.init("ExtClockPicker", clockPicker.resolveWidgetVar(), clientId);
+        encodeClientBehaviors(context, clockPicker);
+
+        wb.attr("placement", clockPicker.getPlacement());
+        wb.attr("align", clockPicker.getAlign());
+        wb.attr("donetext", clockPicker.getDonetext());
+        wb.attr("autoclose", clockPicker.getAutoclose());
+        wb.attr("vibrate", clockPicker.getVibrate());
+
+        wb.finish();
+    }
+
+    protected static String getValueAsString(final FacesContext context, final ClockPicker clockPicker) {
+        final Object submittedValue = clockPicker.getSubmittedValue();
+        if (submittedValue != null) {
+            return submittedValue.toString();
+        }
+
+        final Object value = clockPicker.getValue();
+        if (value == null) {
+            return null;
+        }
+        else {
+            if (clockPicker.getConverter() != null) {
+                // convert via registered converter
+                return clockPicker.getConverter().getAsString(context, clockPicker, value);
+            }
+            else {
+                // use built-in converter
+                SimpleDateFormat timeFormat;
+                timeFormat = new SimpleDateFormat("HH:mm", clockPicker.calculateLocale());
+
+                return timeFormat.format(value);
+            }
+        }
+    }
+
+    @Override
+    public Object getConvertedValue(FacesContext context, UIComponent component,
+                Object value) throws ConverterException {
+        final ClockPicker clockPicker = (ClockPicker) component;
+        String submittedValue = (String) value;
+        SimpleDateFormat format = null;
+
+        if (isValueBlank(submittedValue)) {
+            return null;
+        }
+
+        // Delegate to user supplied converter if defined
+        try {
+            Converter converter = clockPicker.getConverter();
+            if (converter != null) {
+                return converter.getAsObject(context, clockPicker, submittedValue);
+            }
+        }
+        catch (ConverterException e) {
+            throw e;
+        }
+
+        // Delegate to global defined converter (e.g. joda or java8)
+        try {
+            ValueExpression ve = clockPicker.getValueExpression("value");
+            if (ve != null) {
+                Class type = ve.getType(context.getELContext());
+                if (type != null && type != Object.class && type != Date.class) {
+                    Converter converter = context.getApplication().createConverter(type);
```

Comment:
I think you can do `Converter<Object> converter = ComponentUtils.getConverter(context, component);`

Reply:
The implementation has been updated. This line does not exist anymore.